files/common/usr/bin/get-property-from-image

```diff
@@ -80,21 +80,21 @@ UNPACK_DIR=$(mktemp -d -p "$UPDATE_DIR" -t unpack.XXXXXXX)
 [[ -d "$UNPACK_DIR" ]] || die "failed to create unpack directory '$UNPACK_DIR'"
 pushd "$UNPACK_DIR" &>/dev/null || die "'pushd $UNPACK_DIR' failed"
 
-tar -x SHA256SUMS SHA256SUMS.sig.5.3 version.info -f "$UPGRADE_IMAGE_PATH" ||
+tar -x SHA256SUMS SHA256SUMS.sig.* version.info -f "$UPGRADE_IMAGE_PATH" ||
 	die 14 "failed to extract files from upgrade image '$UPGRADE_IMAGE_PATH'"
 
-for file in SHA256SUMS SHA256SUMS.sig.5.3 version.info; do
+for file in SHA256SUMS SHA256SUMS.sig.* version.info; do
 	[[ -f "$file" ]] || die 15 "image is corrupt; missing '$file' file"
 done
 
 if [[ -z "$opt_s" ]]; then
 	openssl dgst -sha256 \
-		-verify /var/lib/delphix-appliance/key-public.pem.upgrade.5.3 \
-		-signature SHA256SUMS.sig.5.3 \
+		-verify /var/lib/delphix-appliance/key-public.pem.upgrade.* \
+		-signature SHA256SUMS.sig.* \
```

Comment:
After adding the `--wildcards` option to the line above that I commented on, this fails:
```
dgst: Can only sign or verify one file.
get-property-from-image: image is corrupt; verification of 'SHA256SUMS' file, using signature 'SHA256SUMS.sig.*' and key 'key-public.pem.upgrade.*' failed
```

I believe this is because there's more than one signature contained in the upgrade image. Specifically, right now, there's the 5.3 and 6.0 signatures.

We'll need to ensure we only use the signature version that matches the private key version that's installed in `/var/lib`.. i.e. if the 6.0 key is installed, we want to verify against the 6.0 signature in the upgrade image (and ignore all others).

I think a good way "fix" this might be to parse the appliance version, to extract the currently running/installed version, store that in a variable early in the script, and then use that variable in the various place that we need it. I can discuss this with you later, if you like to pursue this approach.

Reply:
Here's an example for getting the version:
```
$ zfs get -Hpo value com.delphix:current-version "$(zfs list -Hpo name /)" | cut -d - -f 1 | cut -d . -f 1-2
6.0
```

If we go this route, it might be best to wrap the `zfs` calls in a new script; e.g. `get-appliance-version` (we have a script like this in the app-gate, but IMO, it probably makes most sense to have this in the delphix-platform respository).

Reply:
Thanks @prakashsurya, yes this route seems better, will do.